modulemd/include/private/modulemd-util.h

```diff
@@ -63,49 +129,195 @@ G_DEFINE_AUTOPTR_CLEANUP_FUNC (modulemd_tracer, modulemd_trace_free);
+ * that are nested two levels deep.
+ *
+ * Returns: A newly-allocated #GHashTable containing a deep copy of the
+ * keys from @orig. Each corresponding #GHashTable value is deep copied, with
+ * the second level #GHashTable copied for use as a set as described in
+ * modulemd_hash_table_deep_str_copy().
+ *
+ * Since: 2.0
+ */
 GHashTable *
 modulemd_hash_table_deep_str_str_set_copy (GHashTable *orig);
 
+/**
+ * modulemd_hash_table_sets_are_equal_wrapper:
+ * @a: A void pointer.
+ * @b: A void pointer.
+ *
+ * Returns: TRUE if both @a and @b (considered as #GHashTable<!-- -->s) contain
+ * an identical set of keys, FALSE if they differ.
+ *
+ * Only the keys of @a and @b are compared. The values are ignored.
+ *
+ * Since: 2.2
+ */
 gboolean
 modulemd_hash_table_sets_are_equal_wrapper (const void *a, const void *b);
 
+/**
+ * modulemd_hash_table_sets_are_equal:
+ * @a: A #GHashTable object.
+ * @b: A #GHashTable object.
+ *
+ * Returns: TRUE if both @a and @b contain an identical set of keys, FALSE if
+ * they differ.
+ *
+ * Only the keys of @a and @b are compared. The values are ignored.
+ *
+ * Since: 2.0
+ */
 gboolean
 modulemd_hash_table_sets_are_equal (GHashTable *a, GHashTable *b);
 
+/**
+ * modulemd_hash_table_equals:
+ * @a: A #GHashTable object.
+ * @b: A #GHashTable object.
+ * @compare_func: A #GEqualFunc function that is called to determine the
+ * equivalence of pairs of #GHashTable values.
+ *
+ * Returns: TRUE if both @a and @b contain identical keys and identical
+ * corresponding values (as determined by @compare_func), FALSE if they
+ * differ.
+ *
+ * Since: 2.2
+ */
 gboolean
 modulemd_hash_table_equals (GHashTable *a,
                             GHashTable *b,
                             GEqualFunc compare_func);
 
+/**
+ * modulemd_strcmp_sort:
+ * @a: A #gconstpointer.
+ * @b: A #gconstpointer.
+ *
+ * Returns: 0 if @a and @b are pointers to identical strings, a negative value
+ * if @a is less than @b, and a positive value if @a is greater than @b.
+ *
+ * Since: 2.0
+ */
 gint
 modulemd_strcmp_sort (gconstpointer a, gconstpointer b);
 
+/**
+ * modulemd_ordered_str_keys:
+ * @htable: A #GHashTable.
+ * @compare_func:  A #GCompareFunc function that is called to determine the
+ * equivalence of pairs of #GHashTable keys from @htable.
+ *
+ * Returns: A #GPtrArray of the keys from @htable sorted according to
+ * @compare_func.
+ *
+ * Since: 2.0
+ */
 GPtrArray *
 modulemd_ordered_str_keys (GHashTable *htable, GCompareFunc compare_func);
 
+/**
+ * modulemd_ordered_str_keys_as_strv:
+ * @htable: A #GHashTable.
+ *
+ * Returns: A #GStrv list of the keys from @htable sorted according to
+ * modulemd_strcmp_sort().
+ *
+ * Since: 2.0
+ */
 GStrv
 modulemd_ordered_str_keys_as_strv (GHashTable *htable);
 
+/**
+ * modulemd_variant_deep_copy:
+ * @variant: A #GVariant opaque data structure.
+ *
+ * Returns: A newly-allocated deep copy of @variant.
```

Comment:
```suggestion
 * Returns: (transfer full): A newly-allocated deep copy of @variant.
```

Even though we aren't actually generating bindings for these, it is a helpful clue to developers. It tells them they're required to own the memory once it's passed back.